src/python/pants/backend/python/util_rules/pex.py

```diff
@@ -627,8 +638,15 @@ def _build_pex_description(request: PexRequest) -> str:
                 {', '.join(request.requirements.req_strings)}
                 """
             )
-        elif isinstance(request.requirements.from_superset, LoadedLockfile):
-            lockfile_path = request.requirements.from_superset.lockfile_path
+        elif request.requirements.from_superset is not None:
```

Comment:
You structure it this way twice and I may be missing a logic subtlety or this may just be eye of the beholder, but:
```python
elif if isinstance(request.requirements.from_superset, LoadedLockfile):
  ...
elif isinstance(request.requirements.from_superset, Resolve):
  ...
else:
  assert request.requirements.from_superset is None
```

Seems a bit more clear to me.

Reply:
It was helpful to do it this way in the earlier case (around line 477) because the LoadedLockfile and Resolve cases share a lot of code, the only difference being where the `loaded_lockfile` that code uses comes from. 

Then I did the same thing here for uniformity.

In any case, I should be able to take the LoadedLockfile case away entirely in a followup.

Reply:
Actually, huh, I might be able to do that in this change!

Reply:
Yep!